src/pages/companion/badges/index.tsx

```diff
@@ -34,9 +38,78 @@ const Badges = () => {
   const [badges, setBadges] = useState([]);
   const [completedBadges, setCompletedBadges] = useState(0);
   const [completedHiddenBadges, setCompletedHiddenBadges] = useState(0);
+  const [error, setError] = useState("");
+  const [sortBy, setSortBy] = useState<SortOption>("Name");
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchBadges = async () => {
+      try {
+        setIsLoading(true);
+        const profileId = localStorage.getItem(COMPANION_PROFILE_ID_KEY);
+        if (!profileId) {
+          setError("Please log in to view your connections");
+          setIsLoading(false);
+          return;
```

Comment:
nit: would prefer a redirect to the `/companion` path for them to login, but this should be fine for now - users are likely logged in if they find a way to navigate to `/companion/badges`